src/systems/systems/src/hydraulic/update_iterator.rs

```diff
@@ -283,7 +277,7 @@ mod max_step_tests {
             actual_loop_num += 1;
         }
 
-        //0.320 seconds with max of 0.100 we expect 3 max step duration plus 1 final step so 4
+        //0.320 seconds with max of 0.100 we expect 4 steps
```

Comment:
**thought**: The fact that there is not a single test that changes despite the changes in this file means that more tests should be added for the `MaxFixedStepLoop`.
**suggestion**: Add various tests that confirm the durations returned by the iterator are in fact as you'd expect them to be.

Reply:
I actually check number of loops and final total time, current implementation and previous one gives different delta times but "satisfy" the intitial requirement.  The test is not checking the actual implementation though indeed.